src/APICore.ts

```diff
@@ -1,16 +1,55 @@
 import {backOff} from 'exponential-backoff';
 import {PlatformClientOptions} from './ConfigurationInterfaces.js';
 import getEndpoint, {Environment, Region} from './Endpoints.js';
-import {ResponseHandler} from './handlers/ResponseHandlerInterfaces.js';
-import handleResponse, {defaultResponseHandlers, ResponseHandlers} from './handlers/ResponseHandlers.js';
+import {ResponseHandler} from './Entry.js';
+import handleResponse, {ResponseHandlers} from './handlers/ResponseHandlers.js';
 import {UserModel} from './resources/Users/index.js';
+import {createFetcher} from './utils/createFetcher.js';
 import retrieve from './utils/Retriever.js';
+import {Predicate} from './utils/types.js';
+
+interface TokenInfo {
+    authentication: UserModel;
+    [key: string]: any;
+}
+
+const HEADERS_JSON_CONTENT_TYPE: HeadersInit = Object.freeze({'Content-Type': 'application/json'});
+const FILE_HANDLERS = Object.freeze<ResponseHandler[]>([
+    ResponseHandlers.noContent,
+    ResponseHandlers.successBlob,
+    ResponseHandlers.error,
+]);
+
+/** Check whether the response status is `429 Too Many Requests`. */
+const isTooManyRequests: Predicate<Response> = (response) => response.status === 429;
+/** Check whether the method is 'GET' (case insensitive). */
+const isGet: Predicate<string | undefined> = RegExp.prototype.test.bind(/^GET$/i);
+
+/** "Logical OR" two optional abort signals. */
+const abortOnEither = (
+    signal1: AbortSignal | null | undefined,
+    signal2: AbortSignal | null | undefined
+): AbortSignal | null | undefined => {
+    if (signal1 && signal2) {
+        const joined = new AbortController();
+        function forwardAbort(this: AbortSignal) {
+            joined.abort(this.reason);
+        }
+        signal1.addEventListener('abort', forwardAbort);
+        signal2.addEventListener('abort', forwardAbort);
+        return joined.signal;
+    }
+    return signal1 ?? signal2;
+};
+
+const withBody = (body: any, userArgs: RequestInit | undefined): RequestInit | undefined =>
+    userArgs?.body ? undefined : {headers: HEADERS_JSON_CONTENT_TYPE, body: JSON.stringify(body)};
 
 export default class API {
     static orgPlaceholder = '{organizationName}';
 
     private getRequestsController: AbortController;
-    private tokenInfo: Record<string, any>;
+    private tokenInfo: TokenInfo;
```

Comment:
This signature is wrong, it should be:
```suggestion
    private tokenInfo: TokenInfo | undefined;
```

However, that also affects the signature of (public)
```typescript
    get currentUser(): UserModel {
        return this.tokenInfo?.authentication;
    }
```
Note that the `currentUser` signature is wrong, this will return `undefined` if `checkToken` hasn't been called/didn't complete yet.